azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -100,6 +106,27 @@ def with_aad_application_key_authentication(cls, connection_string, aad_app_id,
         kcsb[kcsb.ValidKeywords.application_key] = app_key
         return kcsb
 
+    @classmethod
+    def with_aad_application_certificate_authentication(
+        cls, connection_string, aad_app_id, certificate, thumbprint
+    ):
+        """Creates a KustoConnection string builder that will authenticate with AAD application and
+        password.
+        :param str connection_string: Kusto connection string should by of the format:
+        https://<clusterName>.kusto.windows.net
+        :param str aad_app_id: AAD application ID.
+        :param str certificate: A PEM encoded certificate private key.

```

Comment:
Huh???

Reply:
https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/dev/adal/authentication_context.py#L237-L247